contracts/puppeteer-initia/src/contract.rs

```diff
@@ -0,0 +1,1161 @@
+
+fn execute_update_config(
+    deps: DepsMut<NeutronQuery>,
+    info: MessageInfo,
+    new_config: ConfigOptional,
+) -> ContractResult<Response<NeutronMsg>> {
+    cw_ownable::assert_owner(deps.storage, &info.sender)?;
+
+    let puppeteer_base = Puppeteer::default();
+    let mut config = puppeteer_base.config.load(deps.storage)?;
+
+    let mut attrs: Vec<Attribute> = Vec::new();
+
+    if let Some(remote_denom) = new_config.remote_denom {
+        config.remote_denom = remote_denom.clone();
+        attrs.push(attr("remote_denom", remote_denom))
+    }
+
+    if let Some(connection_id) = new_config.connection_id {
+        config.connection_id = connection_id.clone();
+        attrs.push(attr("connection_id", connection_id))
+    }
+
+    if let Some(port_id) = new_config.port_id {
+        config.port_id = port_id.clone();
+        attrs.push(attr("port_id", port_id))
+    }
+
+    if let Some(update_period) = new_config.update_period {
+        config.update_period = update_period;
+        attrs.push(attr("update_period", update_period.to_string()))
+    }
+
+    if let Some(allowed_senders) = new_config.allowed_senders {
+        let allowed_senders =
+            validate_addresses(deps.as_ref().into_empty(), allowed_senders.as_ref(), None)?;
+        attrs.push(attr("allowed_senders", allowed_senders.len().to_string()));
+        config.allowed_senders = allowed_senders
+    }
+
+    if let Some(transfer_channel_id) = new_config.transfer_channel_id {
+        config.transfer_channel_id = transfer_channel_id.clone();
+        attrs.push(attr("transfer_channel_id", transfer_channel_id))
+    }
+
+    if let Some(sdk_version) = new_config.sdk_version {
+        config.sdk_version = sdk_version.clone();
+        attrs.push(attr("sdk_version", sdk_version))
+    }
+    if let Some(timeout) = new_config.timeout {
+        validate_timeout(timeout)?;
+        attrs.push(attr("timeout", timeout.to_string()));
+        config.timeout = timeout;
+    }
+
+    puppeteer_base.update_config(deps.into_empty(), &config)?;
+
+    Ok(response("config_update", CONTRACT_NAME, attrs))
+}
+
+fn execute_ibc_transfer(
+    deps: DepsMut<NeutronQuery>,
+    env: Env,
+    info: MessageInfo,
+    reason: IBCTransferReason,
+    reply_to: String,
+) -> ContractResult<Response<NeutronMsg>> {
+    let puppeteer_base = Puppeteer::default();
+    let config = puppeteer_base.config.load(deps.storage)?;
+    validate_sender(&config, &info.sender)?;
+    puppeteer_base.validate_tx_idle_state(deps.as_ref())?;
+    // exclude fees, no need to send local denom tokens to remote zone
+    let message_funds: Vec<_> = info
+        .funds
+        .into_iter()
+        .filter(|f| f.denom != LOCAL_DENOM)
+        .collect();
+    ensure_eq!(
+        message_funds.len(),
+        1,
+        ContractError::InvalidFunds {
+            reason: "Only one coin is allowed".to_string()
+        }
+    );
+    let coin = message_funds.first().ok_or(ContractError::InvalidFunds {
```

Comment:
Can use .unwrap() here, since previous check guarantees that .first() will return Some().

In fact, this error check is unreachable.